.cargo/config.offline

```diff
@@ -1,3 +1,6 @@
+[build]
+target-dir = "target"
```

Comment:
Does this actually prevent user configs from trampling the value? I presume that e.g. environment variables could still override this.

Reply:
Right, this only prevents a config file from overriding it (and I’ve tested that). I had a second change that I didn’t push, roughly
```make
CARGO_TARGET_DIR ?= target

LIBRUSTZCASH=$(top_builddir)/$(CARGO_TARGET_DIR)/$(RUST_TARGET)/release/librustzcash.a
[...]
```
What this was intended to do was use the user’s `CARGO_TARGET_DIR`, but still override `build.target-dir` to “target” in the config. As I alluded to in the commit message, it’s difficult to reference the user’s config file setting outside of Cargo, but easy to use the env var, so this was a compromise.

However, I actually kept getting failures in Cargo itself whenever I set `CARGO_TARGET_DIR` to a different location, which made it basically impossible to test if that change was doing the right thing, so I left that alone.

I’d much rather have a change that just universally accepted the user’s setting, but I don’t see how.